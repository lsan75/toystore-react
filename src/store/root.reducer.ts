import { combineReducers } from 'redux';
import { toyReducer } from './toy/toy.reducer';

export const rootReducer = combineReducers({
  toyReducer
});
