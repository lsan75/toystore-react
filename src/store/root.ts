import { ToyReducerState, defaultToyReducer } from './toy/toy.reducer'
import { AuthReducerState, defaultAuthReducer } from './auth/auth.reducer'

export interface Store {
  toyReducer: ToyReducerState
  authReducer: AuthReducerState
}

export const defaulStore: Store = {
  toyReducer: defaultToyReducer,
  authReducer: defaultAuthReducer
}
