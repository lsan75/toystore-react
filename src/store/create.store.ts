import { createStore, applyMiddleware } from 'redux'
import thunkMiddleware from 'redux-thunk'
import { rootReducer } from './root.reducer'
import { defaulStore } from './root'
import { composeWithDevTools } from 'redux-devtools-extension'

// configure redux
export const store = createStore(
  rootReducer,
  defaulStore,
  composeWithDevTools(applyMiddleware(
    thunkMiddleware
  ))
)
