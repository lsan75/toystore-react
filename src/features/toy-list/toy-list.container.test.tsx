import * as React from 'react';
import { configure, shallow, ShallowRendererProps } from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';
import ToyListContainer from './toy-list.container';
import { Provider } from 'react-redux';
import { store } from '../../store/create.store';

configure({ adapter: new Adapter() });

describe('ToyListContainer', () => {

  const props: ShallowRendererProps = {
    context: { store }
  };

  it('renders without crashing', () => {
    const output = shallow(
      <Provider store={store}>
        <ToyListContainer />
      </Provider>
    ).dive(props);

    expect(shallowToJson(output)).toMatchSnapshot();

  });

});
