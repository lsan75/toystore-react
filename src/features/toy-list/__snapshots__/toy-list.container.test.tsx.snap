// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToyListContainer renders without crashing 1`] = `
<ToyListContainer
  getToys={[Function]}
  select={[Function]}
  toyList={
    Array [
      Object {
        "icon": "ballon",
        "price": 10,
        "title": "toy",
      },
    ]
  }
  unselect={[Function]}
/>
`;
